backend/app/agents/metabase_agent.py

```
@@ -1,56 +1,202 @@
+            METABASE_METADATA_CACHE.clear()
+            METABASE_METADATA_CACHE.update(metadata)
+            logger.info(f"Fetched and cached Metabase metadata: {len(metadata['databases'])} databases.")
+    except Exception as e:
+        logger.error(f"Error fetching Metabase metadata: {e}", exc_info=True)
+
+# Startup: fetch metadata and cache it
+def _startup_metadata_cache():
+    try:
+        loop = asyncio.get_event_loop()
+        if loop.is_running():
+            asyncio.ensure_future(fetch_metabase_metadata())
```

Ruff (0.12.2)

Store a reference to the return value of `asyncio.ensure_future`

(RUF006)